scripts/export_event_types.py

```diff
@@ -45,71 +77,119 @@
     e_ranges.sort()
     log_e_reco_bins = np.log10(event_types.extract_energy_bins(e_ranges))
+    model = trained_model[next(iter(trained_model))]
+    suffix = model[next(iter(model))]["test_data_suffix"]
 
     # Energy binning (in log10 TeV) used to separate event types. We use the binning usually used in
     # sensitivity curves, extended to lower and higher energies.
     event_type_log_e_bins = np.arange(-1.7, 2.5, 0.2)
+    # Camera offset binning (in degrees) used to separate event types. The binning is a test,
+    # should be changed for better performance.
+    event_type_offset_bins = np.arange(0, 7, 1)
+    # event_type_offset_bins = np.append(event_type_offset_bins, 10)
     # Number of event types we want to classify our data:
     n_types = 3
     # This variable will store the event type partitioning container.
     event_type_partition = None
 
-    for dl2_file in dl2_file_list:
-        print("Exporting file: {}".format(dl2_file))
-        dtf = event_types.load_dtf(dl2_file.replace(".root", ""))
+    for particle in particles:
+        print("Exporting files: {}".format(particle))
+        dtf = event_types.load_dtf(particle)
         print("Total number of events: {}".format(len(dtf)))
 
+        # We only bin in energy here to have the same file as in the training and be able to have
+        # the exact same training/testing events' separation.
         dtf_e = event_types.bin_data_in_energy(dtf, log_e_reco_bins=log_e_reco_bins)
 
         # We only separate training statistics in case of exporting a gamma_cone file.
-        if "gamma_cone" in dl2_file:
-            # Using a constant seed of 777, same as in the training/testing events
+        if particle is gamma:
+            # Using a constant seed of 777, same as in the training/testing events.
             dtf_e_train, dtf_e_test = event_types.split_data_train_test(dtf_e, random_state=777)
         else:
             dtf_e_test = dtf_e
 
-        if "gamma" in dl2_file:
-            # To match the format needed by partition_event_types
-            dtf_e_test_formatted = {"default": dtf_e_test}
-            # Add the predicted Y_diff to the data frame:
-            dtf_test = event_types.add_predict_column(dtf_e_test_formatted, trained_model)
+        # To match the format needed by partition_event_types
+        dtf_e_test_formatted = {suffix: dtf_e_test}
+        # Add the predicted Y_diff to the data frame:
+        dtf_test = event_types.add_predict_column(dtf_e_test_formatted, trained_model)
+
+        if particle is gamma:
             # Divide the Y_diff distributions into a discrete number of event types (n_types)
-            d_types, event_type_partition = event_types.partition_event_types(
-                dtf_test,
-                labels=labels,
-                log_e_bins=event_type_log_e_bins,
-                n_types=n_types,
-                return_partition=True,
-            )
+            if offset_partition:
```

Comment:
This `if` can be put within `partition_event_types` and there call either `partition_event_types_energy` or `partition_event_types_energy_and_offset` based on the value of `offset_partition`.